Tests: Make CertificateTest enzyme 3 ready and fix minecraft certificate bug

islemaster opened the pull request:
Following [these instructions](https://docs.google.com/document/d/1D-5CR0OrExDZEsiMJ6obkRFoqp835GO2LEuCCkz8SCs/edit) and reviewing [this list](https://docs.google.com/spreadsheets/d/1xWHtfLmfUmBPsq-vJ44fEucqMG64N0RfLAhCBXjDrTQ/edit#gid=0) I'm fixing up a few more tests to be ready for the Enzyme 3 upgrade.

I believe these are fixed in a backwards-compatible way; letting Drone confirm that.

It turns out these tests weren't actually asserting anything, because `expect(false)` is not a failing assertion!  (Note to self: Find a linter for this.) So I fixed the assertions, which revealed a failing test, so I fixed the bug too.

I considered removing the second test case entirely because the `type` prop went away in https://github.com/code-dot-org/code-dot-org/pull/19240/files. However, I think it still captures a valid case with the `isMinecraft` prop overriding the tutorial setting.  Let me know if I'm missing context here.

islemaster commented on apps/test/unit/templates/CertificateTest.js:
```diff
@@ -7,30 +7,35 @@ import responsive from '@cdo/apps/code-studio/responsiveRedux';
+
+  afterEach(() => {
+    window.dashboard = storedWindowDashboard;
+  });
```
This set/restore allows this test to run in isolation with `--entry`.

Erin007 commented on apps/test/unit/templates/CertificateTest.js:
```diff
@@ -7,30 +7,35 @@ import responsive from '@cdo/apps/code-studio/responsiveRedux';
+      context: {store}
+    }).dive();
+    expect(wrapper.find('img').html()).to.include(
+      'MC_Hour_Of_Code_Certificate'
```
"new" and "older" in these tests are unclear to me at this point because "new" relative to when? when the test were written?  

If I recall correctly, the distinction for certificates is that the Aquatic certificate is blue but all the other Minecraft versions are the same. I think the tests we want are : 
1.) renders a generic Minecraft certificate regardless of tutorial (no tutorial, minecraft any version except aquatic, or any other tutorial) if `isMinecraft={true}`
 2.) renders an aquatic specific certificate if `isMinecraft={true}` && `tutorial='aquatic` =>  'MC_Hour_Of_Code_Certificate_Aquatic'

islemaster replied:
An excellent point!  Since I included a behavior change here, I'll update with the suggested tests before merge.  Thanks for catching this!

islemaster commented:
On further review, the `isMinecraft` prop was always 100% determined by the `tutorial` prop, so I dropped it.  Added tests to cover each certificate case.
